Add new experimentalSetExplicitDimensions prop

polishq opened the pull request:
#### What problem is this solving?

Current code does not allow explicit width and height properties to be set on `img` elements. This PR adds a new `experimentalSetExplicitDimensions` prop which, when set to `true`, will apply the specified `width` and `height` to the `img` tag. The intention is to allow stores to improve their Cumulative Layout Shift score.

#### How to test it?

Linked in this workspace: http://arthur--eriksbikeshop.myvtex.com/
(inspect the 6 category images near the top of the homepage)

Commit:
Add new experimentalSetExplicitDimensions prop

danzanzini reviewed (commented):
Hey @polishq. Have two questions:

- Do this prop work with the `experimentalPreventLayoutSift` prop?
- There is a [height and width properties in the image-list and images array](https://github.com/vtex-apps/store-image#image-list-props). Do they work well together?

Commit:
Remove units from width and height props

Commit:
Add support for new prop to ImageList

polishq commented:
> Hey @polishq. Have two questions:
> 
> * Do this prop work with the `experimentalPreventLayoutSift` prop?
> * There is a [height and width properties in the image-list and images array](https://github.com/vtex-apps/store-image#image-list-props). Do they work well together?

@danzanzini 
1. Yes, the `experimentalPreventLayoutShift` prop adds the width and height as CSS attributes, the new prop adds them as `img` attributes. The two props can be used in conjunction or on their own. I added some logic to strip the units from the width and height prop values (so when the `img` attribute is set, "300px" will become "300"). 
2. Yep, and I've just added support for the prop at the `ImageList` level

This is something that one of our top US clients has been asking for, so our plan is to let their developer validate the solution before we let other clients know about it.

danzanzini commented on react/Image.tsx:
```diff
@@ -128,6 +135,15 @@ function Image(props: ImageProps) {
       style={imageDimensions}
       ref={imageRef}
       className={handles.imageElement}
+      {...(experimentalSetExplicitDimensions &&
+      !width?.toString().includes('%') &&
+      !height?.toString().includes('%') &&
+      (widthWithoutUnits || heightWithoutUnits)
```
Maybe creating a const will clarify this conditional purpose

polishq replied:
Done!

danzanzini reviewed (approved):
Approved. Left one suggestion. 
Can you add the prop to the readme: https://github.com/vtex-apps/store-image#image-list-props

Commit:
Update docs and move conditional logic to const
